filestore/file_readers.py

```diff
@@ -11,7 +11,71 @@
 logger = logging.getLogger(__name__)
 
 
-class _HdfMapsHandlerBase(HandlerBase):
+class _HDF5HandlerBase(HandlerBase):
+
+    def open(self):
+        if self._file:
+            return
+        self._file = h5py.File(self._filename)
+
+    def close(self):
+        super(_HDF5HandlerBase, self).close()
+        self._file.close()
+
+
+class HDF5DatasetSliceHandler(_HDF5HandlerBase):
+    """
+    Handler for data stored in one Dataset of an HDF5 file.
+
+    Parameters
+    ----------
+    filename : string
+        path to HDF5 file
+    key : string
+        key of the single HDF5 Dataset used by this Handler
+    frame_per_point : integer, optional
+        number of frames to return as one datum, default 1
+    """
+    def __init__(self, filename, key, frame_per_point=1):
+        self._fpp = frame_per_point
+        self._filename = filename
+        self._key = key
+        self._file = None
+        self._dataset = None
+        self.open()
+
+    def __call__(self, point_number):
+        # Don't read out the dataset until it is requested for the first time.
+        if not self._dataset:
```

Comment:
To be picky. this does not actually read the dataset into memory, it just sets up the objects to make sure that you can access the data.